lib/bolt/result.rb

```diff
@@ -1,69 +1,71 @@
 require 'json'
+require 'bolt/error'
 
 module Bolt
   class Result
-    attr_reader :message
+    attr_reader :message, :error
 
-    def initialize(message)
-      @message = message
+    def self.from_exception(exception)
+      @exception = exception
+      if @exception.is_a?(Bolt::Error)
+        error = @exception.to_h
+      else
+        error = {
+          'kind' => 'puppetlabs.tasks/exception-error',
+          'issue_code' => 'EXCEPTION',
+          'msg' => exception.message,
+          'details' => { 'class' => exception.class.to_s }
+        }
+        error['details']['stack_trace'] = exception.backtrace.join('\n') if exception.backtrace
+      end
+      Result.new(error)
     end
 
-    def value
-      nil
+    def initialize(error = nil, message = nil)
+      @error = error
+      @message = message
     end
 
-    def error
+    def value
       nil
     end
 
     def to_h
-      { 'value' => value }
-    end
-
-    def to_result
-      value
-    end
-
-    def success?
-      true
-    end
-  end
-
-  class ErrorResult < Result
-    def initialize(message, issue_code, kind)
-      super(message)
-      @issue_code = issue_code
-      @kind = kind
-    end
-
-    def to_h
-      {
-        'error' => {
-          'issue_code' => @issue_code,
-          'kind' => @kind,
-          'msg' => @message
-        }
-      }
+      h = {}
+      if value
+        h['value'] = value
+        h['value']['_output'] = message if message
+      elsif message
+        h['value'] = { '_output' => message }
+      end
+      h['error'] = error if error
+      h
     end
 
     def to_result
-      {
-        '_error' => {
-          'issue_code' => @issue_code,
-          'kind' => @kind,
-          'msg' => @message
-        }
-      }
+      # TODO: This should be to_h but we need to update the plan functions to
```

Comment:
Is there a PUP ticket related to that? Or a BOLT ticket (now that we moved run_* to Bolt)?

Reply:
unfortunately ExecutionResult has a from_bolt method that has to change.

Reply:
That's do-able, just curious what the plan was to do it.